### PR TITLE
[CALCITE-3072] Generate right SQL for FLOOR&SUBSTRING functions in SparkSqlDialect

### PR DESCRIPTION
**Pull request by DonnyZone**
Rewrite `FLOOR` and `SUBSTRING` functions to generate SparkSql-specific query.
`FLOOR->DATE_TRUNC`
`SUBSTRING('Hello World' FROM 5) -> SUBSTRING('Hello World', 5) `
`SUBSTRING('Hello World' FROM 5 FOR 1) -> SUBSTRING('Hello World', 5, 1) `

### CODE REVIEW

**Comment by ambition119** [comment]
@DonnyZone 
You can
```shell
git config --global user.email "your github email" 
```
then Your commit avatar will show highlights.

**Comment by DonnyZone** [comment]
@ambition119 thanks for reminder

File: core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java [hunk 1]
```diff
@@ -55,6 +70,32 @@ public SparkSqlDialect(SqlDialect.Context context) {
...
+
+  @Override public void unparseCall(SqlWriter writer, SqlCall call,
+      int leftPrec, int rightPrec) {
+    if (call.getOperator() == SqlStdOperatorTable.SUBSTRING) {
+      SqlUtil.unparseFunctionSyntax(SPARKSQL_SUBSTRING, writer, call);
+    } else {
+      switch (call.getKind()) {
+      case FLOOR:
+        if (call.operandCount() != 2) {
+          super.unparseCall(writer, call, leftPrec, rightPrec);
```
Comment (danny0405) [inline core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java]: Cause your logic has call operands as `!=2` and `=2`, we need to add test cases for them separately.

Comment (DonnyZone) [reply core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java]: Done. This rewriting logic is similar to that in `PostgresqlSqlDialect` for [CALCITE-1798](https://issues.apache.org/jira/browse/CALCITE-1798). Add a test case for one operand.

**Comment by danny0405** [comment]
+1, LGMT

**Pull request closed without merging**